apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template-inserter.php

```diff
@@ -98,6 +98,31 @@ public function fetch_template_parts() {
 		}
 	}
 
+	/**
+	 * Retries a call to wp_remote_get on error.
+	 *
+	 * @param string $request_url Url of the api call to make.
+	 * @param string $request_args Addtional arguments for the api call.
+	 * @param int    $attempt The number of the attempt being made.
+	 * @return array wp_remote_get reponse array
+	 */
+	private function fetch_retry( $request_url, $request_args = null, $attempt = 1 ) {
+		$max_retries = 3;
+
+		$response = wp_remote_get( $request_url, $request_args );
+
+		if ( ! is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		if ( $attempt > $max_retries ) {
+			return;
+		}
+
+		$attempt++;
```

Comment:
What do you think about adding `sleep` with exponential backoff here while keeping the retries at 3?
E.g. 
```php
sleep( pow( 2, $attempt ) )
```

If there are intermittent network or API problems, things are less likely to recover in a short interval burst of three successive requests without delays.

It is blocking the theme switch and plugin activate operation a bit, but that's done only once in a while anyway.

Reply:
See also https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer/wpcom-http/pipeline/retry-on-failure#delay-calculation

Where @dmsnell mapped this out and had fun with graphs.

Reply:
Good idea - have added this in, and if this is done before the $attempt++ gives a max of 14 sec wait if all attempts fails

Reply:
Just wanted to say that "@dmsnell had fun with graphs" made me shiver a little bit 😨